docs/data/date-pickers/custom-opening-button/CustomOpeningIconConditional.tsx

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import { Dayjs } from 'dayjs';
+import { createSvgIcon } from '@mui/material/utils';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { DemoContainer } from '@mui/x-date-pickers/internals/demo';
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+
+const PriorityHighIcon = createSvgIcon(
```

Comment:
What do you think about just using the `@mui/icons-material`?
The first demo already showcased having custom SVGs, others could cater to a different audience. 🤔

Reply:
Oh right, in the doc we can use `icons-material`, I'll update it :+1:

Reply:
> And even on the 1st demo because I think roughly nobody uses our createSvgIcon utility function.

I wouldn't mind keeping one demo with custom solutions.
We had at least one issue, AFAIK, asking how to use custom SVG files or/and paths. 🤔

Reply:
How about rendering 3 pickers on the 1st demo

1. SVG from icons-material
2. SVG from file
3. SVG manually rendered

Reply:
Looks great. 👍 
It's maybe a shame that we loose the code preview in such case. 🙈 🤷

Reply:
Do you know why we loose it?
Because I agree that it's very usefull here...

Reply:
![image](https://github.com/mui/mui-x/assets/3309670/1483f7d3-83de-4ff3-ab11-9214a560ed6e)

I changed the formating to be below this limit